### PR TITLE
Data store

### PR DESCRIPTION
**Pull request by lukasberglund**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,27 +27,47 @@
...
+    Entity commentEntity = new Entity("Comment");
+
+    commentEntity.setProperty("content", content);
+
+    return commentEntity;
+  }
+
+  private void storeEntity(Entity entity) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(entity);
```
Comment (prabenzom) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: Does this generate a unique id per comment? How do we differentiate between comments with the same text in the db?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,27 +27,47 @@
...
+  }
+
+  private String getComment(HttpServletRequest request) {
+    String comment = request.getParameter("new-comment");
+
+    return comment;
+  }
+
+  private Entity buildCommentEntity (String content) {
+    Entity commentEntity = new Entity("Comment");
```
Comment (prabenzom) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: This should probably be a constant?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,27 +27,47 @@
...
-  
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-    
-    comments.add("Love your website!!"); 
-    comments.add("Your face is funny"); 
-    comments.add("Hot singles in your area!!! Visit obvious-spam-site9809.com!!!"); 
-  }
+  private List<String> comments = new ArrayList<>();
```
Comment (prabenzom) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: are you planning on making the comment object more complex than a string?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,27 +27,47 @@
...
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (prabenzom) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: Does the rest of the walkthrough go through List vs Get? I'd imagine eventually we'd want this get function to return a single comment and for list to return all comments.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (lukasberglund) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:32]: Yes!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (prabenzom) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: I just took a look at the documentation for Http servlet. It looks like we'd need a separate path for list vs get. So I think we can leave this as is.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (lukasberglund) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: It does. Here is a screenshot of the result when I write the same thing in a comment twice https://screenshot.googleplex.com/2dnWuBb5tnY. I'm gonna add data like timestamp and maybe username later on but I think it's good for now.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (lukasberglund) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: What exactly should be a constant? The return value, the parameter, or the `commentEntity`?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,27 +27,47 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String content = getComment(request);
+
+    Entity commentEntity = buildCommentEntity(content);
+    storeEntity(commentEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  private String getComment(HttpServletRequest request) {
```
Comment (alygerharter) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: i think this extra function is unnecessary - just use request.getParameter above (line 44)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (alygerharter) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: you can improve this in a future PR.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (lukasberglund) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

**Pull request merged**